### PR TITLE
Adds timeouts to requests get/post/put methods

### LINKED ISSUE #367: Add timeouts in all requests to external APIs
## Problem Definition

There are many places in the codebase where there are requests made in external apis without a set timeout. This can lead to rotkehlchen hanging.

## Task

Add timeouts to all requests and handle the cases where timeouts do occur. Docs: https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts

### PR DESCRIPTION
**Pull request by yabirgb**
Also modifies the timeout for beaconchain allowing slower responses 
Closes #367

### CODE REVIEW

**Review by LefterisJP — COMMENTED** [review]
As you can see from all the failing tests you will also have to edit the mock methods of the tests to also include a timeout argument. (sorry) ^_^

**Comment by yabirgb** [comment]
Yes I'm on it :D

File: docs/changelog.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Changelog
 * :feature:`2844` Premium users will now be able to fetch their Uniswap v3 swaps.
 * :feature:`2893` Users can now see if any of their addresses have CVX available to claim from the ConvexFinance airdrop.
 * :bug:`2850` User will now see a consistent naming of exchanges across the application.
+* :bug:`367` Requests to external services won't make rotki hang for long periods.
```
Comment (LefterisJP) [inline docs/changelog.rst:19]: This does not really need a changelog. But if it does, then this one sounds like every request to external service may hang for a long period. Perhaps change to: 
```suggestion
* :bug:`367` Fixed edge cases where some tasks would run for hours due to the absence of timeouts.
```

File: rotkehlchen/constants/timing.py [hunk 2]
```diff
@@ -16,4 +16,6 @@
 
 DEFAULT_CONNECT_TIMEOUT = 5
 DEFAULT_READ_TIMEOUT = 30
+BEACONCHAIN_READ_TIMEOUT = 75
 DEFAULT_TIMEOUT_TUPLE = (DEFAULT_CONNECT_TIMEOUT, DEFAULT_READ_TIMEOUT)
+BEACONCHAIN_TIMEOUT_TUPLE = (DEFAULT_CONNECT_TIMEOUT, BEACONCHAIN_READ_TIMEOUT)
```
Comment (LefterisJP) [inline rotkehlchen/constants/timing.py:21]: Since this is specific to beacon chain only keep it in the beaconchain.py file

**Review by LefterisJP — COMMENTED** [review]
Some small comments. Other than that looks good to me. Did not check though if it's all the places that would need changing.

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**